tools/src/lib.rs

```diff
@@ -113,6 +113,22 @@ impl Address {
     }
 }
 
+impl Default for Address {
```

Comment:
As far as I can tell, you don't need to implement `Default` by hand here, you could just `derive` it.

Reply:
For some reason I though floats didn't implement `Default` :D